lib/chef/resource.rb

```diff
@@ -1090,18 +1088,15 @@ def self.action(action, description: nil, &recipe_block)
     # @param action [Symbol,String] the action name
     # @return the description of the action provided, or nil if no description
     # was defined
-    def self.action_description(action)
-      action_descriptions[action.to_sym]
-    end
-
-    # @api private
-    #
-    # @return existing action description hash, or newly-initialized
-    # hash containing action descriptions inherited from parent Resource,
-    # if any.
-    def self.action_descriptions
-      @action_descriptions ||=
-        superclass.respond_to?(:action_descriptions) ? superclass.action_descriptions.dup : { nothing: nil }
```

Comment:
this bit down here would work in the provider as well wouldn't it?

we don't care about dynamic late inheritance, and we can assume the parent class is fully built and all the descriptions are there by the time we are constructing the subclass.

Reply:
yeah i convinced myself this would work, then i convinced myself it wouldn't work, now i'm back to convincing myself it would work, i think my mental simulator is broken.  if it works, this'd be simpler though.